README.md

```diff
@@ -1,76 +1,124 @@
 
-### Step 5: Configure your web app using `PublicClient.java`
+> In the next steps, you might need the tenant name (or directory name) or the tenant ID (or directory ID). These are presented in the **Properties**
+  of the Azure Active Directory window respectively as *Name* and *Directory ID*
 
-Enter the client id value above at the top of the `PublicClient.java` file.
+#### Register the app app (Native-Headless-Application)
 
-```java
-public class PublicClient {
+1. In the  **Azure Active Directory** pane, click on **App registrations** and choose **New application registration**.
+1. Enter a friendly name for the application, for example 'Native-Headless-Application' and select 'Native' as the *Application Type*.
+1. For the *Redirect URI*, enter `https://<your_tenant_name>/Native-Headless-Application`, replacing `<your_tenant_name>` with the name of your Azure AD tenant.
+1. Click on **Create** to create the application.
+1. In the succeeding page, Find the *Application ID* value and copy it to the clipboard. You'll need it to configure the Visual Studio configuration file for this project.
+1. Then click on **Settings**, and choose **Properties**.
+1. Configure Permissions for your application. To that extent, in the Settings menu, choose the 'Required permissions' section and then,
+   click on **Add**, then **Select an API**, and type `Microsoft Graph` in the textbox. Then, click on  **Select Permissions** and select **User.Read**.
 
-    private final static String AUTHORITY = "https://login.microsoftonline.com/common/";
-    private final static String CLIENT_ID = "<your client id>";
+### Step 4:  Configure the sample to use your Azure AD tenant
 
-```
+In the steps below, ClientID is the same as Application ID or AppId.
 
-### Step 6: Package and then run the `public-client-adal4j-sample-jar-with-dependencies.jar ` file.
+#### Configure the app project
+
+1. Open the `src\main\java\PublicClient.java` file
+1. Find the app key `private final static String CLIENT_ID` and replace the existing value with the application ID (clientId) of the `Native-Headless-Application` application copied from the Azure portal.
+
+### Step 5: Run the sample
 
 From your shell or command line:
 
-* `$ mvn package`
+- `$ mvn package`
 
 This will generate a `public-client-adal4j-sample-jar-with-dependencies.jar` file in your /targets directory. Run this using your Java executable like below:
 
-* `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
-
+- `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
 
 ### You're done!
 
-Your command line interface should prompt you for the username and password and then access the Graph API to retreive your user information.
+Your command line interface should prompt you for the username and password and then access the Microsoft Graph API to retrieve your user information.
+
+## Community Help and Support
```

Comment:
I suggest you have a section "About the code", where you explain that the relevant code is line 45-47 ? (and maybe add it in a ```Java block

Reply:
Done